rules/manager_test.go

```diff
@@ -193,6 +195,168 @@ func TestAlertingRule(t *testing.T) {
+		"HTTPRequestRateLow",
+		expr,
+		time.Minute,
+		0,
+		labels.FromStrings("severity", "{{\"c\"}}ritical"),
+		labels.EmptyLabels(), labels.EmptyLabels(), "", false, nil,
+	)
+
+	recordingRule := NewRecordingRule(
+		"HTTPRequestRate",
+		expr,
+		labels.EmptyLabels())
+
+	var ruleGroupActive atomic.Bool
+	ruleGroupActive.Store(true)
+
+	evalRuleGroupFunc := func(ctx context.Context, g *Group, evalTimestamp time.Time, doneCh <-chan struct{}) bool {
+		return ruleGroupActive.Load()
+	}
+
+	iterations := 0
+	var shouldMissEval atomic.Bool
+	shouldMissEval.Store(false)
+	evalIterationFunc := func(ctx context.Context, g *Group, evalTimestamp time.Time) {
+		iterations++
+		if shouldMissEval.Load() {
+			time.Sleep(2000 * time.Millisecond)
+			DefaultEvalIterationFunc(ctx, g, evalTimestamp)
+			return
+		}
+		modifiedTS := time.Unix(0, 0).Add(time.Duration(iterations) * time.Second)
+		DefaultEvalIterationFunc(ctx, g, modifiedTS)
+	}
+
+	registry := prometheus.NewRegistry()
+
+	opts := &ManagerOptions{
+		ExternalURL:     nil,
+		QueryFunc:       EngineQueryFunc(testEngine, storage),
+		NotifyFunc:      func(ctx context.Context, expr string, alerts ...*Alert) {},
+		Context:         context.Background(),
+		Appendable:      storage,
+		Queryable:       storage,
+		Logger:          log.NewNopLogger(),
+		OutageTolerance: 30 * time.Minute,
+		ForGracePeriod:  10 * time.Minute,
+		Registerer:      registry,
+		Metrics:         nil,
+	}
+
+	group := NewGroup(GroupOptions{
+		Name:              "active-passive",
+		Interval:          time.Second,
+		Rules:             []Rule{newRule, recordingRule},
+		ShouldRestore:     true,
+		Opts:              opts,
+		EvalIterationFunc: evalIterationFunc,
+		EvalRuleGroupFunc: evalRuleGroupFunc,
+	})
+
+	gatherMetrics := func() (map[string]float64, error) {
+		ms, err := registry.Gather()
+		if err != nil {
+			return nil, err
+		}
+		result := make(map[string]float64)
+		for _, m := range ms {
+			for _, mn := range metricNames {
+				if m.GetName() == mn {
+					if m.GetType() == io_prometheus_client.MetricType_COUNTER {
+						for _, k := range m.GetMetric() {
+							result[m.GetName()] = k.Counter.GetValue()
+						}
+					}
+					if m.GetType() == io_prometheus_client.MetricType_GAUGE {
+						for _, k := range m.GetMetric() {
+							result[m.GetName()] = k.Gauge.GetValue()
+						}
+					}
+				}
+			}
+		}
+		return result, nil
+	}
+
+	ctx := context.Background()
+	go func(ctx context.Context) {
+		group.run(ctx)
+	}(ctx)
+
+	time.Sleep(2 * time.Second)
+	require.False(t, group.GetLastEvalTimestamp().IsZero())
+	require.False(t, group.ShouldRestore())
+	metrics, err := gatherMetrics()
+	require.NoError(t, err)
+	require.Equal(t, float64(4), metrics[metricNames[0]])
+	require.Equal(t, float64(2), metrics[metricNames[1]])
+	require.Equal(t, float64(0), metrics[metricNames[2]])
+	require.Equal(t, float64(2), metrics[metricNames[3]])
+
+	// deactivate
+	ruleGroupActive.Swap(false)
```

Comment:
Can we assert that the rule did not evaluate when we switched it to false?

Reply:
Done